store/postgres/src/block_store.rs

```diff
@@ -462,6 +462,28 @@ impl BlockStore {
 
         Ok(())
     }
+
+    pub fn truncate_block_caches(&self) -> Result<(), StoreError> {
+        for (_chain, store) in &*self.stores.read().unwrap() {
+            store.truncate_block_cache()?
+        }
+        Ok(())
+    }
+
+    pub fn update_db_version(&self) -> Result<(), StoreError> {
+        use crate::primary::db_version as dbv;
+        use diesel::prelude::*;
+
+        let connection = self.mirror.primary().get()?;
```

Comment:
You really want the primary here, not the mirror. The mirror reads from other shards if the primary is down, but `db_version` is not copied to other shards; so something like `self.pools.get(&*PRIMARY_SHARD).unwrap().get()?`